### PR TITLE
FRI-566 - Added processing of diff files

### PR DESCRIPTION
**Pull request by astro-snail**

### CODE REVIEW

File: reporting-engine-worker/src/main/java/org/ihtsdo/termserver/scripting/reports/release/PackageComparisonReport.java [hunk 1]
```diff
@@ -119,7 +124,7 @@ public Job getJob() {
 	}
 
 	@Override
-	protected void loadProjectSnapshot(boolean fsnOnly) throws TermServerScriptException, InterruptedException, IOException {
+	protected void loadProjectSnapshot(boolean fsnOnly) {
 	}
```
Comment (pgwilliams) [inline reporting-engine-worker/src/main/java/org/ihtsdo/termserver/scripting/reports/release/PackageComparisonReport.java:130]: I'd put a comment in here to explain why it's necessary to have an empty method.  In this case you'll explain that this process does not need to hold a snapshot in memory, so you're overriding the default behaviour in doing nothing here.

File: reporting-engine-worker/src/main/java/org/ihtsdo/termserver/scripting/reports/release/PackageComparisonReport.java [hunk 2]
```diff
@@ -166,20 +172,30 @@ currentReleaseName,	extractDate(currentReleasePath),
 				if (exitValue != 0) {
 					throw new TermServerScriptException("Script execution failed with exit code: " + exitValue);
 				}
-				info("Script execution finished with exit code: " + exitValue);
+				info("Script execution finished successfully");
```
Comment (pgwilliams) [inline reporting-engine-worker/src/main/java/org/ihtsdo/termserver/scripting/reports/release/PackageComparisonReport.java:177]: Yes well spotted, not possible to be anything other than successful here.

File: reporting-engine-worker/src/main/java/org/ihtsdo/termserver/scripting/reports/release/PackageComparisonReport.java [hunk 3]
```diff
@@ -166,20 +172,30 @@ currentReleaseName,	extractDate(currentReleasePath),
...
 
-			// Write results to Google Sheets
-			report(PRIMARY_REPORT, previousReleasePath, currentReleasePath, "s3://snomed-compares/" + uploadFolder);
-			for (String line: output) {
-				report(SECONDARY_REPORT, line);
+			// Process resulting diff files
+			String path = "scripts/" + uploadFolder + "/target/c";
+
+			try (Stream<Path> stream = Files.list(Paths.get(path))) {
+				stream.filter(file -> !Files.isDirectory(file))
```
Comment (pgwilliams) [inline reporting-engine-worker/src/main/java/org/ihtsdo/termserver/scripting/reports/release/PackageComparisonReport.java:188]: Love a good stream :-)

File: reporting-engine-worker/src/main/java/org/ihtsdo/termserver/scripting/reports/release/PackageComparisonReport.java [hunk 4]
```diff
@@ -196,4 +212,68 @@ private String extractDate(String path) {
...
+				}
+
+				String[] cols = line.substring(2).split("\t", 2);
+
+				String key = cols[0];
+				String value = cols[1];
+
+				switch (ch) {
+					case '<':
+						// Previous release entry
```
Comment (pgwilliams) [inline reporting-engine-worker/src/main/java/org/ihtsdo/termserver/scripting/reports/release/PackageComparisonReport.java:243]: Helpful comments 👍

File: reporting-engine-worker/src/main/java/org/ihtsdo/termserver/scripting/reports/release/PackageComparisonReport.java [hunk 5]
```diff
@@ -196,4 +212,68 @@ private String extractDate(String path) {
...
+		String previousValue;
+		String currentValue;
+
+		ValuePair(String previousValue, String currentValue) {
+			this.previousValue = previousValue;
+			this.currentValue = currentValue;
+		}
+
+		boolean isInactivated() {
+			return "1".equals(previousValue.split("\t")[1]) && "0".equals(currentValue.split("\t")[1]);
```
Comment (pgwilliams) [inline reporting-engine-worker/src/main/java/org/ihtsdo/termserver/scripting/reports/release/PackageComparisonReport.java]: I'd like to see you define some constant here so we know what context [1] is being used in, like DIFF_IDX_ACTIVE or whatever you want to call it.   Trivial preference.

**Review by pgwilliams — APPROVED** [review]
Looks really good.   No concerns here especially since this is greenfield development so there's no danger of regression.    You're welcome to either merge yourself now, or address those very minor feedback points first.  🥇

File: reporting-engine-worker/src/main/java/org/ihtsdo/termserver/scripting/reports/release/PackageComparisonReport.java (hunk 5 above)
Comment (astro-snail) [reply reporting-engine-worker/src/main/java/org/ihtsdo/termserver/scripting/reports/release/PackageComparisonReport.java]: @pgwilliams I agree, I was thinking about using constants more, too. Will make the change, thanks!

File: reporting-engine-worker/src/main/java/org/ihtsdo/termserver/scripting/reports/release/PackageComparisonReport.java (hunk 1 above)
Comment (astro-snail) [reply reporting-engine-worker/src/main/java/org/ihtsdo/termserver/scripting/reports/release/PackageComparisonReport.java:130]: Done

**Pull request merged**